src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkConfig.java

```diff
@@ -5,28 +5,67 @@
 import org.apache.kafka.common.config.ConfigDef;
 import org.apache.kafka.common.config.ConfigDef.Importance;
 import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Width;
+import org.testng.util.Strings;
 
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
 
 public class KustoSinkConfig extends AbstractConfig {
+  
     // TODO: this might need to be per kusto cluster...
-    static final String KUSTO_URL = "kusto.url";
-    static final String KUSTO_TABLES_MAPPING = "kusto.tables.topics_mapping";
+    static final String KUSTO_URL_CONF = "kusto.url";
+    private static final String KUSTO_URL_DOC = "Kusto cluster ingestion URL.";
+    private static final String KUSTO_URL_DISPLAY = "Kusto URL";
 
-    static final String KUSTO_AUTH_USERNAME = "kusto.auth.username";
-    static final String KUSTO_AUTH_PASSWORD = "kusto.auth.password";
-
-    static final String KUSTO_AUTH_APPID = "kusto.auth.appid";
-    static final String KUSTO_AUTH_APPKEY = "kusto.auth.appkey";
-    static final String KUSTO_AUTH_AUTHORITY = "kusto.auth.authority";
-
-    static final String KUSTO_SINK_TEMPDIR = "kusto.sink.tempdir";
-    static final String KUSTO_SINK_FLUSH_SIZE = "kusto.sink.flush_size";
-    static final String KUSTO_SINK_FLUSH_INTERVAL_MS = "kusto.sink.flush_interval_ms";
-    static final String KUSTO_SINK_WRITE_TO_FILES = "kusto.sink.write_to_files";
+    static final String KUSTO_AUTH_USERNAME_CONF = "kusto.auth.username";
+    private static final String KUSTO_AUTH_USERNAME_DOC = "Kusto username for authentication, also configure kusto.auth.password.";
+    private static final String KUSTO_AUTH_USERNAME_DISPLAY = "Kusto Auth Username";
+    
+    static final String KUSTO_AUTH_PASSWORD_CONF = "kusto.auth.password";
+    private static final String KUSTO_AUTH_PASSWORD_DOC = "Kusto password for the configured username.";
+    private static final String KUSTO_AUTH_PASSWORD_DISPLAY = "Kusto Auth Password";
+    
+    static final String KUSTO_AUTH_APPID_CONF = "kusto.auth.appid";
+    private static final String KUSTO_AUTH_APPID_DOC = "Configure this to perform authentication using Kusto AppID, "
+        + "also configure kusto.auth.appkey and kusto.auth.authority.";
+    private static final String KUSTO_AUTH_APPID_DISPLAY = "Kusto Auth AppID";
+    
+    static final String KUSTO_AUTH_APPKEY_CONF = "kusto.auth.appkey";
+    private static final String KUSTO_AUTH_APPKEY_DOC = "Kusto AppKey for authentication.";
```

Comment:
Application key for Azure Active Directory authentication

Reply:
can i ask why did u deprecate the configs : kusto.tables.topics_mapping, kusto.sink.flush_size and kusto.sink.flush_interval_ms?

Reply:
@ohadbitt we usually follow dot-separated configuration names while implementing connectors. It's just a standard that brings uniformity across all the Kafka Connectors(especially the gold connectors). No other specific reason.